### PR TITLE
[AWS] Add EMR metrics data stream

### LINKED ISSUE #6290: [AWS] [EMR] Metrics data stream
## Overview

Amazon EMR, formerly known as Amazon Elastic MapReduce, is an advanced cluster platform designed to simplify, scale, and optimize the processing and analysis of massive data volumes. With its user-friendly interface, flexibility, and affordability, EMR offers an ideal solution for organizations seeking to leverage big data. The platform boasts a wide range of data processing engines, such as Apache Hadoop, Apache Spark, Apache Hive, Apache Flink, and many others, enabling users to harness the power of these tools for efficient data manipulation and insights generation.

AWS EMR clusters send metrics to CloudWatch in 5 minute intervals (1 minute if detailed monitoring is enabled).

Collect EMR Hadoop 2.x metrics from AWS EMR clusters.

## Steps/Tasks

- Setup AWS EMR Cluster
    - Set up an EMR cluster with Hadoop 2.x.
- Fetch metrics using the generic `cloudwatch` metricbeat module
    - Utilize the cloudwatch metricbeat module to fetch the required EMR Hadoop 2.x metrics.
- Create AWS EMR metrics integration
    - Integrate the collected metrics into our existing monitoring infrastructure.
    - Ensure proper handling and processing of EMR Hadoop 2.x metrics.
- Create AWS EMR metrics data stream documentation
    - Create documentation detailing the newly created data stream for EMR Hadoop 2.x metrics.
    - Include comprehensive information about metric names, dimensions, and their meanings.
- Add pipeline & pipeline tests (if needed)
    - Implement a pipeline, if necessary, to process the collected metrics.
    - Develop tests to verify the correctness of the pipeline.
- Add systems tests using terraform
    - Use terraform to create automated systems tests that validate the end-to-end metrics collection.
    - Verify the accurate collection and ingestion of the metrics.

## Docs

Metrics and dimensions that should be collected for Hadoop 2.x clusters:
- https://docs.aws.amazon.com/emr/latest/ManagementGuide/UsingEMR_ViewingMetrics.html

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Adds AWS EMR (ElasticMapReduce) metrics data stream. 

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates #6084 
- Closes #6290

### CODE REVIEW

**Review by ritalwar — COMMENTED** [review]
Could you please define the metric type as either counter or gauge for applicable fields? This would be beneficial from a TSDB perspective in the future.

File: packages/aws/_dev/build/docs/emr.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Amazon EMR (ElasticMapReduce)
+
+The Amazon EMR integration allows you to monitor [Amazon EMR](https://aws.amazon.com/emr/) — a fully managed big data processing and analytics service.
+
+Use the Amazon EMR integration to collect metrics related to your EMR instances. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference the metrics when troubleshooting an issue.
+
+For example, you could use this data to track Amazon EMR cluster progress and cluster storage. Then you can alert when utilization for an instance crosses a predefined threshold.
```
Comment (lucianpy) [inline packages/aws/_dev/build/docs/emr.md:7]: What about clarifying the Hadoop version for this data stream in this doc?

Idea: Creating a meta issue for AWS integrations that we own, to improve documentation clarity. What are your thoughts on this? @zmoog @lalit-satapathy

Comment (zmoog) [reply packages/aws/_dev/build/docs/emr.md:7]: Yeah, this is a good idea. @lucianpy, could you create an issue with a detailed list of information to add to this integration? 

This way, we can ship the integration as it is and start working on an actionable list right after the current iteration ends.

**Review by zmoog — APPROVED** [review]
Looks good, @gpop63! Could you create an issue and add the steps required to move this integration from BETA to GA? Many thanks!

**Pull request merged**